src/govpress-unit/hotfixing.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Hotfixing code in GovPress
+last_reviewed_at: ""
+---
+
+GovPress site repositories follow this pattern:
+
+1. A `develop` branch is protected and will deploy to a staging site on merge.
+1. A `main` branch is protected and will deploy to a production site on merge.
+
+_Hotfixing_ in our context means creating a PR from a branch off `main` and merging into `main`, skipping `develop` altogether.
+
+This is a risky operation because:
+
+1. We have no way of testing the new code in a deployed environment. The code will be checked in the CI/CD pipeline but not against a database of content that is "like" production.
+1. The staging and production sites will now be out of sync, and so staging will no longer be an accurate test for code we wish to deploy to production.
+1. We risk creating a merge conflict between the `develop` and `main` branches which will block future production deploys.
+
+For these reasons, we have a bias towards avoiding hotfix merges as far as possible and we try to keep `develop` branches in a state where they are always deployable.
+
+## Reducing the likelihood of hotfixes
+
+Sometimes we start a long-running piece of development work, for example a redesign of an active theme, which is likely to hold up production deploys.
+
+In this case, we should first consider factoring our code to avoid hotfixes. For example, can new features be designed as a new plugin, which can be deactivated by default? Can a theme rebuild be placed in a new theme?
+
+An alternative to a new plugin or theme is to guard the new feature with a feature flag which can be set to activate the feature in staging but not in production. This is something that is much easier to implement if your code is well factored, according to the [SOLID principles of OOP](https://en.wikipedia.org/wiki/SOLID).
```

Comment:
Non-blocking suggestion: would it be useful to include a link to an explainer about feature flags here? (e.g. https://blog.jetbrains.com/space/2022/06/16/feature-flags/). Some members of the team might not be familiar with the concept.